test/get/git/git_lfs_test.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+@TestOn('linux')
+import 'package:test/test.dart';
+import 'package:test_descriptor/test_descriptor.dart' show sandbox;
+import 'package:path/path.dart' as p;
+import 'package:pub/src/exit_codes.dart' as exit_codes;
+import 'package:pub/src/io.dart' show runProcess;
+
+import '../../descriptor.dart' as d;
+import '../../test_pub.dart';
+
+const pubspec = 'name: foo\r\ndescription: test\r\nversion: 3.0.0';
+const gitattributes = 'foo.dylib filter=lfs diff=lfs merge=lfs -text';
+
+String fakeGit(hash, lfs) {
+  return '''
+#!/bin/bash -e
+case \$@ in
+  --version)
+    echo 'git version 2.16'
+    ;;
+  rev-list*)
+    echo '$hash'
+    ;;
+  show*)
+    echo '$pubspec'
+    ;;
+  checkout*)
+    mkdir -p $sandbox/cache/git/foo-$hash/.git
+    echo '$pubspec' > $sandbox/cache/git/foo-$hash/pubspec.yaml
+    echo '$gitattributes' > $sandbox/cache/git/foo-$hash/.gitattributes
+    ;;
+  $lfs
+esac
+''';
+}
+
+void main() {
+  test('reports failure if Git LFS is needed but is not installed', () async {
+    ensureGit();
+
+    final repo = d.git('foo.git', [
+      d.libDir('foo'),
+      d.file('pubspec.yaml', pubspec),
+      d.file('.gitattributes', gitattributes)
+    ]);
+
+    await repo.create();
+    final ref = await repo.runGit(['rev-list', '--max-count=1', 'HEAD']);
+    final hash = ref.first;
+
+    await d.appDir({
+      'foo': {
+        'git': {'url': '../foo.git'}
+      }
+    }).create();
+
+    const lfs = '''
+  lfs*)
+    echo "git: 'lfs' is not a git command. See 'git --help'."
+    exit 1
+    ;;
+  ''';
+
+    await d.dir('bin', [d.file('git', fakeGit(hash, lfs))]).create();
+    final binFolder = p.join(sandbox, 'bin');
+    // chmod the git script
+    if (!Platform.isWindows) {
+      await runProcess('chmod', ['+x', p.join(sandbox, 'bin', 'git')]);
+    }
```

Comment:
Might it not be wiser to make a set of tests that are skipped unless:
 * (A) git LFS is installed, and
 * (B) we are not running in CI (env var `CI=true`.

Then change the github actions environment to install git lfs, so that the tests will pass.

Reply:
Let me clarify, I think these test cases where we mock LFS is a good idea, but I don't think they are sufficient.

I think we need tests where we have LFS installed and use actual LFS. That ideally also means running an LFS server on localhost.